### DISCUSSION TITLE
Return Leafmap object in flask

### DISCUSSION THREAD

**Discussion by Pranav4399** [opengeos/leafmap#439] [Q&A] [answered]
Hello. I''m trying to integrate leafmap into my flask app where I take latitude and longitude from the user and display the map accordingly. But I'm facing trouble with returning the map object. Attaching the code for your reference below - 

`
@app.route('/coords/<lat>/<lon>')
def coords(lat, lon):
    m = leafmap.Map(center=[lat, lon], zoom=18)
    m.add_basemap('SATELLITE')
    return m
`

The 'return m' line is throwing me error. I'm able to achieve the same functionality with Folium. I'm attaching the code for that as well below - 

`
@app.route('/coords/<lat>/<lon>')
def coords(lat, lon):
    m = folium.Map(location=(lat, lon))
    return m.get_root().render()
`

**Comment by EmanuelCastanho** [comment]
I have the same question, I am using Folium with Flask. Is it possible to use Leafmap with Flask instead?

**Comment by giswqs** [comment, marked as answer]
I don't use flask. For building interactive web apps, I recommend solara. Folium is very limited. 

https://huggingface.co/spaces/giswqs/solara-geospatial
